### ISSUE TITLE
Incompatible with react-leaflet v3

### ISSUE THREAD

**Issue by victorlin** [vadzim/use-leaflet#109]
Runtime error:

> useLeafletMap.js:10 Uncaught TypeError: (0 , _reactLeaflet.useLeaflet) is not a function
    at useLeafletMap (useLeafletMap.js:10)
    ...

According to [this migration guide](https://sean-rennie.medium.com/migrating-react-leaflet-from-v2-to-v3-12d6088af191):

> No more `useLeaflet` hook or `withLeaflet` HOC. It’s been replaced with `useMap` instead

Downgrading to `react-leaflet` v2 fixes the issue. However, v2 is no longer maintained.

Related: PaulLeCam/react-leaflet#783

**Comment by hugobarragon** [comment]
I think it does not make sense any more on v3 because o offical support
https://react-leaflet.js.org/docs/api-map/#usemap

**Comment by vadzim** [comment]
> I think it does not make sense any more on v3 because o offical support https://react-leaflet.js.org/docs/api-map/#usemap

v2 also have a hook. `useLeaflet` has some other hooks, so for me it still makes sense))

**Comment by hugobarragon** [comment]
yes, thats true @vadzim I this lib has other useful hooks, on our case that we only need the useMap, that react-leaflet has it too maybe it is not needed
